includes/attendee/attendee.php

```diff
@@ -37,7 +37,20 @@ public function is_host(): bool {
 		return $this->is_host;
 	}
 
+	/**
+	 * Check if the user is the unique host of the event.
+	 *
+	 * @return bool
+	 */
+	public function is_unique_host(): bool {
+		return ( new Attendee_Repository() )->is_unique_host( $this->event_id, $this->user_id );
```

Comment:
I think that when using the repository pattern, the "entity" classes (`Event`, `Attendee`, etc) don't typically call other repositories, they are only "containers" for data, and their methods operate solely on that data.

So, if you would want to have a `is_foo()` method, the value of `foo` would need to be passed to the constructor, and then that method would `return $this->foo;`.

However, in this case, I don't think `is_unique_host` should be passed to the constructor, because its value depends on data that does not "belong" to this attendee (it depends on other attendees leaving the event), so you could be in a situation where `is_unique_host()` returns false when it should in fact return `true`, because an attendee left after the instance of `Attendee` was created.

I think we could remove the `is_unique_host()` from the `Attendee`, and call it on the repository instead:

```php
// attend-event.php

if ( $this->attendee_repository->is_unique_host( $event_id, $user_id ) ) {
    // Can't remove last host
}
```

Alternatively, because what we want to know is whether the event has a single attendee, we could instead use the already existing `get_hosts()` method:

```php
if ( 1 === count( $this->attendee_repository->get_hosts( $event_id ) ) ) {
    // Can't remove last host
}
```

This would mean we could remove the `is_unique_host()` from the repository, and from `Attendee`.

Reply:
Thank you for the feedback. Done!